bitcoin/src/blockdata/script/push_bytes.rs

```diff
@@ -43,34 +44,38 @@ mod primitive {
     pub struct PushBytes([u8]);
 
     impl PushBytes {
-        /// Creates `&Self` without checking the length.
+        /// Creates `&PushBytes` without checking the length.
         ///
         /// # Safety
         ///
         /// The caller is responsible for checking that the length is less than the [`LIMIT`].
         unsafe fn from_slice_unchecked(bytes: &[u8]) -> &Self {
+            // SAFETY: The caller must guarantee that bytes.len() < 0x100000000.
+            // If that is the case the conversion is safe because &[u8] and &PushBytes
+            // have the same layout (because of #[repr(transparent)] on PushBytes).
             &*(bytes as *const [u8] as *const PushBytes)
         }
```

Comment:
I feel like this could be a public function (from_mut_slice_unchecked too)

Reply:
Maybe at some point but I don't want to spend time figuring out if the name is reasonable right now.